client/v1/status.go

```diff
@@ -0,0 +1,46 @@
+package v1
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/leptonai/gpud/manager/packages"
+)
+
+// GetPackageStatus fetches the GPUd package status from the GPUd admin API.
+func GetPackageStatus(ctx context.Context, url string) ([]packages.PackageStatus, error) {
+	httpClient := &http.Client{
```

Comment:
Why not use options here?  It's recommended that if we need to customize http client, it's better to define a type `Client` represents the gpud client, and refactor the function as a Client's method. So that we can just focus on the   core logic in the function.

Reply:
Done